src/test/java/eu/dissco/core/translator/terms/utils/AgentUtilsTest.java

```diff
@@ -0,0 +1,94 @@
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class AgentUtilsTest {
+
+  public static Stream<Arguments> agentProvider() {
+    return Stream.of(
+        Arguments.of(null, null, List.of()
+        ),
+        Arguments.of("Fricke, Ronald", "http://orcid.org/0000-0002-9079-593X",
+            List.of(createAgent("Fricke, Ronald", "http://orcid.org/0000-0002-9079-593X"))
+        ),
+        Arguments.of("Fricke, Ronald", null,
+            List.of(createAgent("Fricke, Ronald", null))
+        ),
+        Arguments.of("http://orcid.org/0000-0002-9079-593X", null,
+            List.of(createAgent(null, "http://orcid.org/0000-0002-9079-593X"))
+        ),
+        Arguments.of("http://orcid.org/0000-0002-9079-593X",
+            "https://orcid.org/0000-0001-9790-9277",
+            List.of(createAgent("http://orcid.org/0000-0002-9079-593X",
+                "https://orcid.org/0000-0001-9790-9277"))
+        ),
+        Arguments.of("  Fricke, Ronald  | Tom Dijkema  ",
+            "http://orcid.org/0000-0002-9079-593X |  https://orcid.org/0000-0001-9790-9277  ",
+            List.of(
+                createAgent("Fricke, Ronald", "http://orcid.org/0000-0002-9079-593X"),
+                createAgent("Tom Dijkema", "https://orcid.org/0000-0001-9790-9277"))
+        ),
+        Arguments.of("  Fricke, Ronald  & https://orcid.org/0000-0001-9790-9277  ", null,
+            List.of(
+                createAgent("Fricke, Ronald", null),
+                createAgent(null, "https://orcid.org/0000-0001-9790-9277"))
+        ),
+        Arguments.of("Fricke, Ronald",
+            "http://orcid.org/0000-0002-9079-593X |  https://orcid.org/0000-0001-9790-9277",
+            List.of(
+                createAgent(null, "http://orcid.org/0000-0002-9079-593X"),
+                createAgent(null, "https://orcid.org/0000-0001-9790-9277"))
+        ),
+        Arguments.of("Fricke, Ronald & Tom Dijkema",
+            "https://orcid.org/0000-0001-9790-9277",
+            List.of(
+                createAgent("Fricke, Ronald", null),
+                createAgent("Tom Dijkema", null))
+        )
+    );
+  }
+
+  private static Agent createAgent(String name, String id) {
+    var agent = new Agent()
+        .withId(id)
+        .withType(SCHEMA_PERSON)
+        .withSchemaName(name)
+        .withSchemaIdentifier(id)
+        .withOdsHasRoles(
+            List.of(new eu.dissco.core.translator.schema.OdsHasRole().withType("schema:Role")
+                .withSchemaRoleName(CREATOR.getName())));
+    if (id != null) {
+      agent.withOdsHasIdentifiers(
+          List.of(new eu.dissco.core.translator.schema.Identifier().withType("ods:Identifier")
+              .withId(id).withDctermsIdentifier(id)));
+    }
+    return agent;
+  }
+
+  @ParameterizedTest
+  @MethodSource("agentProvider")
+  void testSimpleAgent(String termValue, String id, List<Agent> expected) {
+    // Given
+    var result = new ArrayList<Agent>();
```

Comment:
This isn't a good representation of what actually happens in this class. You don't pass an ArrayList, you pass a List. 
e.g. in the baseDigitalObjectDirector, you'd pass `citation.getOdsHasAgents()`, or `etityRelationship.getOdsHasAgents()`, which returns a List. 

I think you need to create a new ArrayList from the `XX.getOdsHasAgents()` result, then set the agent list to the output of `setAgent()` function.

Reply:
Renaming it to agentValue and agentId

Reply:
Ok made an update here so that it will also handle it well with an unmutable List. This wouldn't really cause an issue because the underlying implementation is always an ArrayList but who knows. I also like that the method really outputs something instead of secretly modifies.